File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap3/WhileIf.py
#!/usr/bin/env python

### While_If.py

print "Enter numbers, one per line.  When done, just enter return.";#(A)

sum = 0                                                             #(B)
while 1:                                                            #(C)
    num = raw_input().rstrip()                                      #(D)
    if num == '':                                                   #(E)
        print "Sum of numbers is: ", sum                            #(F)
        break                                                       #(G)
    sum += float( num )                                             #(H)

File: projects/demos/engineering.purdue.edu/scriptingwithobjects/swocode/chap2/WhileIf.pl
#!/usr/bin/perl -w                                                 

## WhileIf.pl

use strict;                                                        

print "Enter numbers, one per line. When done, just enter return\n";#(A)

my $sum;                                                            #(B)

while (<>) {                                                        #(C)
    if (/^\s*$/) {                                                  #(D)
        printf "Sum of numbers is: %d\n", $sum;                     #(E)
        last;                                                       #(F)
    }                                                               #(G)
    $sum += $_;                                                     #(H)
}
